registry/interface.go

```diff
@@ -39,6 +39,8 @@ type Registry interface {
 	SetUnitTargetState(name string, state job.JobState) error
 	SetMachineState(ms machine.MachineState, ttl time.Duration) (uint64, error)
 	UnscheduleUnit(name, machID string) error
+	UpdateMachineMetadata(machID string, key string, value string) error
```

Comment:
Let's call this `SetMachineMetadata`, since `Update` can mean the key must already exist in some contexts.

And there's no need to respecify `string` three times, just use `SetMachineMetadata(machID, key, value string)`